lib/keyrod/cli.rb

```diff
@@ -122,22 +122,31 @@ def validate_config_group(options, group, message)
     end
 
     def process_tokens
-      if Keyrod::Settings[:'refresh-token']
-        oidc_client = Keyrod::OIDCClient.new
-        Keyrod::Settings[:'access-token'] = oidc_client.access_token
-      end
+      access_token if Keyrod::Settings[:'refresh-token']
+
       fedcloud_client = Keyrod::FedcloudClient.new
       unscoped_token = fedcloud_client.unscoped_token
       projects = fedcloud_client.projects(unscoped_token)
-      if Keyrod::Settings[:group] && !projects.include?(Keyrod::Settings[:group])
-        unless Keyrod::Settings[:'interactive-fallback']
-          raise Keyrod::Errors::ProjectError, "Group #{Keyrod::Settings[:group]} is not available"
-        end
-        Keyrod::Settings[:group] = nil
-      end
-
+      check_projects(projects)
       project = Keyrod::Settings[:group] ? Keyrod::Settings[:group] : ask('Choose one of these groups:', limited_to: projects)
+
       $stdout.puts fedcloud_client.scoped_token(unscoped_token, project)
+    rescue Keyrod::Errors::ProjectError => e
+      abort e.message
+    end
+
+    def access_token
+      oidc_client = Keyrod::OIDCClient.new
+      Keyrod::Settings[:'access-token'] = oidc_client.access_token
+    end
+
+    def check_projects(projects)
```

Comment:
Yes, separating the code into multiple methods is certainly a good idea but I had in mind more like changing a flow of the code so it will be more obvious what's happening and when.

Reply:
I'm not sure how much it can be simplified, as in `token` method the settings are all set and in `process_tokens` method the actual requests are executed. I suppose I can merge `token` and `process_tokens` to keep it straightforward, but I don't see any point in that. I feel like `process_tokens` is simple enough, maybe put it straight below `token` in class?